libcxx/include/chrono

```diff
@@ -1061,7 +1061,7 @@ private:
     rep __rep_;
 public:
 
-#if !defined(_LIBCUDACXX_CXX03_LANG)
+#if !defined(_LIBCUDACXX_CXX03_LANG) && !defined(__NVCC__)
```

Comment:
This line is suspect, but required in order to declare `__shared__` durations in kernels with nvcc <= 10.2.

Reply:
Please explain? This is counterintuitive to me, did it not understand `= default` as trivial when appropriate?

Anyway if anything, to enable this, this should be a version check and not just blank disable of `= default` for all versions of NVCC.

Reply:
Here's an example of a line that errors (with CUDA 10.2, haven't tried 11 or < 10.2):

[`  __shared__ T temp_data[block_size];`](https://github.com/rapidsai/cudf/blob/c39e9168c6d08236f36e5c3203b3ba8d145e86ed/cpp/include/cudf/detail/copy_if.cuh#L108)

```
../../../../include/cudf/detail/copy_if.cuh (108): error: initializer not allowed for __shared__ variable 
```